app/Http/Controllers/Auth/AuthController.php

```diff
@@ -122,140 +125,236 @@ public function create(array $data)
+                return errorResponse(__('message.otp_verification.max_attempts_exceeded'));
+            }
+
+            $attempts->mobile_attempt = (int) $attempts->mobile_attempt + 1;
+            $attempts->save();
+
+            if (! $this->sendOtp($user->mobile_code.$user->mobile)) {
+                return errorResponse(__('message.otp_verification.send_failure'));
+            }
+
+            return successResponse(__('message.otp_verification.send_success'));
+        } catch (\Exception $e) {
+            return errorResponse(__('message.otp_verification.send_failure'));
         }
     }
 
     public function retryOTP(Request $request)
     {
-        $this->validate($request, [
-            'code' => 'required|numeric',
-            'mobile' => 'required',
-        ]);
+        $default_type = $request->input('default_type');
+
+        return match ($default_type) {
+            'email' => $this->sendEmail($request, 'GET'),
+            'mobile' => $this->resendOTP($request),
+        };
+    }
 
+    public function resendOTP($request)
+    {
+        $request->validate([
+            'eid' => 'required|string',
+            'type' => 'required|string|in:text,voice',
+        ]);
         try {
-            $code = $request->code;
-            $mobile = ltrim($request->mobile, '0');
-            $formatted_mobile = sprintf('(+%s) %s', $code, $mobile);
+            $email = Crypt::decrypt($request->eid);
+            $type = $request->input('type');
 
-            $result = $this->sendForReOtp($mobile, $code, $request->type);
+            $user = User::where('email', $email)->firstOrFail();
 
-            $response = ['type' => 'success'];
-            $response['message'] = match ($request->type) {
-                'text' => 'OTP has been resent to '.$formatted_mobile.'.Please Enter the OTP to login!!',
-                default => 'Voice call has been sent to '.$formatted_mobile.'.Please Enter the OTP received on the call to login!!',
-            };
+            if (! $user) {
+                return errorResponse(__('message.user_does_not_exist'));
+            }
 
-            return $response;
-        } catch (\Exception $e) {
-            return response()->json([
-                'response' => $e->getMessage(),
-            ], 500);
+            // Handle verification attempts
+            $attempts = VerificationAttempt::firstOrNew(['user_id' => $user->id]);
+
+            $attempts->save();
+
+            if ($attempts->updated_at->addDay() < Carbon::now()) {
+                $attempts->mobile_attempt = 0;
+                $attempts->email_attempt = 0;
+                $attempts->save();
+            }
+
+            if ($attempts->mobile_attempt >= 3) {
+                return errorResponse(__('message.otp_verification.resend_max_attempts_exceeded'));
+            }
+
+            $attempts->mobile_attempt = (int) $attempts->mobile_attempt + 1;
+            $attempts->save();
```

Comment:
use increment()

Reply:
the column type is varchar in db, so can't use increment